docs/design/design_v1.9.md

```diff
@@ -69,196 +67,246 @@ This means we aim to standardize:
+ - If kubeadm is invoked with `--feature-gates=DynamicKubeletConfig`,  the corresponding feature on kube-apiserver is activated with `--feature-gates=DynamicKubeletConfig=true` flag.
 
 Other flags that are set unconditionally:
 
- - `insecure-port` to `0
- - The `BootstrapTokenAuthenticator` authentication module is enabled using the `--enable-bootstrap-token-auth` flag
- - `--client-ca-file` to `ca.crt`
- - `--tls-cert-file` to `apiserver.crt`
- - `--tls-private-key-file` to `apiserver.key`
- - `--kubelet-client-certificate` to `apiserver-kubelet-client.crt`
- - `--kubelet-client-key` to `apiserver-kubelet-client.key`
- - `--service-account-key-file` to `sa.pub`
- - `--requestheader-client-ca-file` to `front-proxy-ca.crt`
- - `--admission-control` to `Initializers, NamespaceLifecycle, LimitRanger, ServiceAccount, PersistentVolumeLabel, DefaultStorageClass, DefaultTolerationSeconds, NodeRestriction, ResourceQuota`...or whatever the recommended set of admission controllers is at a given version
- - `--kubelet-preferred-address-types` to `InternalIP,ExternalIP,Hostname;` this makes `kubectl logs` and other apiserver -> kubelet communication work in environments where the hostnames of the nodes aren't resolvable
+ - `--insecure-port=0` to avoid insecure connections to the api server.
+ - `--enable-bootstrap-token-auth=true` to enable the `BootstrapTokenAuthenticator` authentication module. 
+ - `--allow-privileged` to `true` (used e.g. by kube proxy).
+ - `--client-ca-file` to `ca.crt`.
+ - `--tls-cert-file` to `apiserver.crt`.
+ - `--tls-private-key-file` to `apiserver.key`.
+ - `--kubelet-client-certificate` to `apiserver-kubelet-client.crt`.
+ - `--kubelet-client-key` to `apiserver-kubelet-client.key`.
+ - `--service-account-key-file` to `sa.pub`.
+ - `--requestheader-client-ca-file` to `front-proxy-ca.crt`.
+ - `--admission-control` to:
+    - [ `Initializers`]() to enable [Dynamic Admission Control](https://kubernetes.io/docs/admin/extensible-admission-controllers/).
+    - [`NamespaceLifecycle`](https://kubernetes.io/docs/admin/admission-controllers/#namespacelifecycle) to avoid deletion of system reserved namespaces etc.
+    - [`LimitRanger`](https://kubernetes.io/docs/admin/admission-controllers/#limitranger) and [`ResourceQuota`](https://kubernetes.io/docs/admin/admission-controllers/#resourcequota) to enforce limits on namespaces
+    - [`ServiceAccount`](https://kubernetes.io/docs/admin/admission-controllers/#serviceaccount) to enforce service account automation.
+    - [`PersistentVolumeLabel`](https://kubernetes.io/docs/admin/admission-controllers/#persistentvolumelabel)  attaches region or zone labels to PersistentVolumes as defined by the cloud provider. Nb. (this admission controller is included for v1.8 clusters or for clusters with cloudProvider `aws` or `gce`).
```

Comment:
(Note: This admission controller is deprecated and will be removed in a future version. It is not deployed by kubeadm by default with v1.9 onwards when not explicitely opting into using `gce` or `aws` as cloud providers.)